include/zephyr/net/ethernet.h

```diff
@@ -132,7 +136,7 @@ struct net_eth_addr {
  * Extend the max frame size for DSA (KSZ8794) by one byte (to 1519) to
  * store tail tag.
  */
-#if defined(CONFIG_NET_DSA_DEPRECATED)
+#if defined(CONFIG_NET_DSA)
```

Comment:
So we could avoid this ifdef if in dsa_core.h we could define `DSA_TAG_SIZE` to 0 if DSA is not enabled. This would make the code more readable.

Reply:
Right. Actually both dsa.h and dsa_core.h already defined ``DSA_TAG_SIZE`` by checking CONFIG_DSA_TAG_SIZE.
So, let me remove the ifdef directly.